web/Services/PropertyService.cs

```diff
@@ -125,24 +131,21 @@ public async Task UpdateProperty(int propertyId, PropertyViewModel updateModel,
 
         // Occupied state takes precedence over attempted update
         // If no update, fallback to current value as usual
-        dbModel.Availability = dbModel.OccupiedUnits == dbModel.TotalUnits
-            ? AvailabilityState.Occupied
-            : updateModel.Availability ?? dbModel.Availability;
-
-        if (dbModel.Availability == AvailabilityState.AvailableSoon)
+        if (dbModel.OccupiedUnits == dbModel.TotalUnits)
         {
-            if (updateModel.Availability == AvailabilityState.AvailableSoon)
-            {
-                // If update succeeds in making the property "available soon", then use its from date
-                dbModel.AvailableFrom = updateModel.AvailableFrom;
-            }
+            dbModel.Availability = AvailabilityState.Occupied;
         }
         else
         {
-            // If we're not "available soon" then don't have a from date
-            dbModel.AvailableFrom = null;
+            dbModel.Availability = updateModel.Availability == AvailabilityState.AvailableSoon
+                ? AvailabilityState.Available
+                : updateModel.Availability ?? AvailabilityState.Draft;
         }
 
+        dbModel.AvailableFrom = dbModel.Availability == AvailabilityState.Available
+            ? updateModel.Availability == AvailabilityState.AvailableSoon? updateModel.AvailableFrom : DateTime.Now
+            : null;

```

Comment:
still a lot of nested ternaries here, which may be worth expanding

Reply:
This nested ternary is directly equivalent to the one in the object initialiser in AddNewProperty, so I don't think it makes sense to expand this one without also expanding the other one (it would disguise their similarities); if I expand both of them, then I believe the one in AddNewProperty would have to move out of the object initialiser. I'm not sure that the benefits of expanding the ternary are significant enough to justify that disadvantage, although if you strongly think that it should be expanded, it can be.